math/sum_of_digits.go

```diff
@@ -0,0 +1,17 @@
+package math
+
+// SumOfDigits returns sum of digits of the integer
+func SumOfDigits(num int64) (sum int64) {
```

Comment:
```suggestion
func SumOfDigits[T constraints.Integer](num T) (sum T) {
```

This applies to all `Integers` not just `int64` 😄